ckanext/dia/harvester/dcat.py

```diff
@@ -76,15 +79,25 @@ def _fetch_license_id(self, license_url):
         licenses = license_list({'model': model}, {})
         try:
             resp = requests.get(license_url)
-        except Exception:
-            log.exception(
-                "Failed to get on license url: {}".format(license_url))
+        except (requests.exceptions.InvalidSchema,
+                requests.exceptions.InvalidURL,
+                requests.exceptions.MissingSchema):
+            self._save_object_error(
+                f'License URL is invalid.',
+                harvest_object)
+            return None
+        except Exception as e:
+            log.exception("Failed to fetch license.")
+            self._save_object_error(
+                f"Fetching license at '{license_url}' failed: {e}",
+                harvest_object)
             return None
         # dealing with direct CC url don't call for a json response
         try:
-            if "https://creativecommons.org" in license_url:
+            if urlparse(license_url).netloc == 'creativecommons.org':
                 for license in licenses:
-                    if license['url'] == resp.url and resp.status_code == 200:
+                    if (urlparse(license['url'])._replace(scheme='http') ==
+                       urlparse(resp.url)._replace(scheme='http') and resp.status_code == 200):
```

Comment:
What's the `_replace(scheme='http')` needed for?

Reply:
The purpose is to ensure that 'https' and 'http' is matched correctly. Its a bit broad brush because technically this would make any protocol match i.e. 'ftp://creativecommons.org/...' but I'm not sure that's something we need to be concerned about...

Reply:
Fair :+1: